.npmignore

```diff
@@ -0,0 +1,12 @@
+.editorconfig
+.eslintrc.js
+.github
+benchmark
+bin/.eslintrc.json
+browser
+docs
+examples
+lib/.eslintrc.json
+src
+test
+tools
```

Comment:
I don't think CHANGELOG.md should ship, but I'm willing to be talked out of that.  I dislike that the .eslintrc's in the subdirs are in different format than the base one, but we can clean that up later.